runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -350,20 +370,35 @@ static int    registered_heap_info_set;
 static size_t registered_heap_size;
 static void*  registered_heap_start;
 
-#define NUM_MEM_REGIONS 10
+static int    using_hugepages;
+static size_t hugepage_size;
+
+//
+// Memory regions.  mem_regions contains the address/length pairs and
+// uGNI memory domain handles for each memory region of interest to us.
+// mem_region_map contains a copy of every node's mem_regions.
+//
+#define MAX_MEM_REGIONS 100
```

Comment:
Yeah, this could definitely be set at execution time.  I'd prefer to do that as part of post-commit work, though.  I'll add it to the list for that, and I'll do the post-commit issue right after merging this.